packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1679,14 +1699,46 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
-    public async createDataStore(pkg: string | string[]): Promise<IFluidRouter> {
-        return this._createDataStore(pkg, false /* isRoot */);
+    private async createDataStoreCore(
+        pkg: string | string[],
+        isRoot: boolean,
+        id: string,
+        props?: any,
+    ): Promise<IFluidDataStoreChannel> {
+        const fluidDataStore = await this._createDataStore(pkg, isRoot, id, props);
+        if (isRoot) {
+            fluidDataStore.bindToContext();
+        }
+
+        return fluidDataStore;
+    }
+
+    public async createDataStore(pkg: string | string[]): Promise<IDataStore> {
+        const internalId = uuid();
+        return channelToDataStore(
+            await this.createDataStoreCore(pkg, false /* isRoot */, internalId),
+            internalId,
+            this,
+            this.mc.logger);
+    }
+
+    private async createAndAliasDataStore(pkg: string | string[], alias: string, props?: any): Promise<IDataStore> {
```

Comment:
I'd expect API not to allow assigning of multiple aliases to same data store.
And as result - APIs not to return IDataStore interface for data stores that already have alias assigned.